src/api/app/controllers/webui/watched_items_controller.rb

```diff
@@ -35,6 +38,12 @@ def set_item
             end
   end
 
+  def set_watchable
+    @package = Package.find_by_project_and_name(params[:project_name], params[:package_name])
+    @project = Project.find_by(name: params[:project_name])
```

Comment:
```suggestion
    @project = Project.get_by_name(params[:project_name])
```

Reply:
think about [links](https://github.com/openSUSE/open-build-service/wiki/Links)....

Reply:
One of the cases I tested by chance: we have a project named [link_project](https://obs-reviewlab.opensuse.org/rubhanazeem-watchlist-ajax/project/show/linked_project) linked to another one, as its meta contains `<link project="home:Admin"/>`. So it has a bunch of inherited packages. I go to the overview of [one of the inherited packages](https://obs-reviewlab.opensuse.org/rubhanazeem-watchlist-ajax/package/show/linked_project/ad) and try to add it to the Watchlist, however, we are not displaying the "Watch this package" link there. In this corner case, we are safe.

Reply:
@hennevogel it's updated. We did some more tests and found that we do not allow the user to watch the linked package. This [check](https://github.com/openSUSE/open-build-service/blob/a03fb07c37cf3f040d7447054f76a866a7230f9f/src/api/app/components/watchlist_component.rb#L35) does that work.

Reply:
Okay but you still will have to use the other methods...